javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithParameters.java

```diff
@@ -133,30 +134,47 @@ default Parameter addAndGetParameter(Parameter parameter) {
+     * then it will be considered as a match only if the given string is {@code "List&lt;String&gt;"}, but not if the
+     * given string is only {@code "List"}.
      *
-     * @param paramTypes the types of parameters like "Map&lt;Integer,String&gt;","int" to match<br> void
-     * foo(Map&lt;Integer,String&gt; myMap,int number)
-     * @return true if all parameters match
+     * @param paramTypes the types of parameters like {@code "Map&lt;Integer,String&gt;", "int"} to match
+     *                   {@code void foo(Map&lt;Integer,String&gt; myMap, int number)}.
+     * @return {@code true} if all parameters match one by one, in the given order.
      */
     default boolean hasParametersOfType(String... paramTypes) {
         return getParameters().stream()
-                .map(p -> p.getType().toString())
-                .collect(toSet())
-                .equals(Stream.of(paramTypes).collect(toSet()));
+                .map(p -> p.getType().asString())
+                .collect(toList())
+                .equals(Stream.of(paramTypes).collect(toList()));
     }
 
     /**
-     * Check if the parameters have certain types. Note that this is a match in SimpleName, so "java.awt.List" and
-     * "java.util.List" are identical to this algorithm.
+     * Check if the parameters have certain types. Note that this is a match in SimpleName, so {@code java.awt.List} and
+     * {@code java.util.List} are identical to this algorithm. In addition, note that it is the erasure of each type
+     * which is considered, so passing {@code List.class} to this method will be considered a match if this node has
+     * exactly one parameter whose type is named {@code List}, regardless of whether the parameter type is declared
+     * without generics as {@code List}, or with generics as {@code List&lt;String&gt;}, or {@code List&lt;Integer&gt;},
+     * etc.
      *
-     * @param paramTypes the types of parameters like "Map&lt;Integer,String&gt;","int" to match<br> void
-     * foo(Map&lt;Integer,String&gt; myMap,int number)
-     * @return true if all parameters match
+     * @param paramTypes the types of parameters like {@code Map.class, int.class} to match
+     *                   {@code void foo(Map&lt;Integer,String&gt; myMap, int number)}.
+     * @return {@code true} if all parameters match one by one, in the given order.
      */
     default boolean hasParametersOfType(Class<?>... paramTypes) {
-        return getParameters().stream().map(p -> p.getType().toString())
-                .collect(toSet())
-                .equals(Stream.of(paramTypes).map(Class::getSimpleName).collect(toSet()));
+        return getParameters().stream()
+                // if p.getType() is a class or interface type, we want to consider its erasure, i.e., if the parameter
+                // is "List<String>", we want to consider it as "List", so we need to call getName()
+                .map(p -> p.getType() instanceof ClassOrInterfaceType
```

Comment:
There's a nice shorthand for that: https://javaparser.org/less-casting-in-javaparser/

Reply:
Oh nice, I didn't know about those `to...()` methods, it's a nice trick. :-)